int32 RAISE=1 
int32 LOWER=-1

int32 value
